test/arborist/build-ideal-tree.js

```diff
@@ -22,12 +22,24 @@ const warningTracker = () => {
   }
 }
 
+const normalizePath = path => path.replace(/[A-Z]:/, '').replace(/\\/g, '/')
+const normalizePaths = obj => {
+  for (const key in obj) {
+    if (key === 'location') {
+      obj[key] = normalizePath(obj[key])
+    } else if (typeof obj[key] === 'object' && obj[key] !== null) {
+      obj[key] = normalizePaths(obj[key])
+    }
+  }
+  return obj
+}
+
 // two little helper functions to make the loaded trees
 // easier to look at in the snapshot results.
 const printEdge = (edge, inout) => ({
   name: edge.name,
   type: edge.type,
-  spec: edge.spec,
+  spec: normalizePath(edge.spec),
```

Comment:
All the other `normalizePath` calls seem reasonable, but shouldn't `edge.spec` always be unixy for `file:` specifiers here, if it's not absolute?  (For the `resolved` field, that's a different story of course, since it is pulling from `fetchSpec`.)

Reply:
workspaces use symlinks, which in windows are junctions, which are absolute paths. because of that the `{CWD}` replacement that takes place doesn't work because a windows absolute path leads with the drive letter and contains the wrong slashes, so we have to normalize it to allow the `{CWD}` replacement to work